flytekit/control_plane/launch_plan.py

```diff
@@ -0,0 +1,287 @@
+import uuid as _uuid
+from typing import Any, List
+
+import six as _six
+
+from flytekit.common.exceptions import scopes as _exception_scopes
+from flytekit.common.exceptions import user as _user_exceptions
+from flytekit.common.mixins import launchable as _launchable_mixin
+from flytekit.configuration import sdk as _sdk_config
+from flytekit.control_plane import identifier as _identifier
+from flytekit.control_plane import interface as _interface
+from flytekit.control_plane import nodes as _nodes
+from flytekit.control_plane import notifications as _notifications
+from flytekit.control_plane import workflow_execution as _workflow_execution
+from flytekit.engines.flyte import engine as _flyte_engine
+from flytekit.models import common as _common_models
+from flytekit.models import execution as _execution_models
+from flytekit.models import identifier as _identifier_model
+from flytekit.models import interface as _interface_models
+from flytekit.models import launch_plan as _launch_plan_models
+from flytekit.models import literals as _literal_models
+
+
+class FlyteLaunchPlan(
+    _launchable_mixin.LaunchableEntity,
+    _launch_plan_models.LaunchPlanSpec,
+):
+    def __init__(self, *args, **kwargs):
+        super(FlyteLaunchPlan, self).__init__(*args, **kwargs)
+        # Set all the attributes we expect this class to have
+        self._id = None
+
+        # The interface is not set explicitly unless fetched in an engine context
+        self._interface = None
+
+    @classmethod
+    def promote_from_model(cls, model: _launch_plan_models.LaunchPlanSpec) -> "FlyteLaunchPlan":
+        return cls(
+            workflow_id=_identifier.Identifier.promote_from_model(model.workflow_id),
+            default_inputs=_interface_models.ParameterMap(
+                {
+                    k: _promises.Input.promote_from_model(v).rename_and_return_reference(k)
```

Comment:
this is old right?  let's nix this.  can we not use the parameters from the model directly?